cmd/cluster-agent/main.go

```diff
@@ -0,0 +1,186 @@
+		Short: "Print the version number",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			av, _ := version.New(version.AgentVersion)
+			fmt.Println(fmt.Sprintf("Cluster Agent from Agent %s - Codename: %s - Commit: %s - Serialization version: %s", av.GetNumber(), av.Meta, av.Commit, serializer.AgentPayloadVersion))
+		},
+	}
+
+	confPath string
+)
+
+// run the host metadata collector every 14400 seconds (4 hours)
+const hostMetadataCollectorInterval = 14400
+
+func init() {
+	// attach the command to the root
+	clusterAgentCmd.AddCommand(startCmd)
+	clusterAgentCmd.AddCommand(versionCmd)
+
+	// local flags
+	startCmd.Flags().StringVarP(&confPath, "cfgpath", "c", "", "path to datadog.yaml")
+
+	config.Datadog.BindPFlag("conf_path", startCmd.Flags().Lookup("cfgpath"))
+}
+
+func start(cmd *cobra.Command, args []string) error {
+	config.Datadog.SetConfigFile(config.Datadog.GetString("conf_path"))
+	confErr := config.Datadog.ReadInConfig()
+
+	// Setup logger
+	syslogURI := config.GetSyslogURI()
+	logFile := config.Datadog.GetString("log_file")
+	if config.Datadog.GetBool("disable_file_logging") {
+		// this will prevent any logging on file
+		logFile = ""
+	}
+	err := config.SetupLogger(
+		config.Datadog.GetString("log_level"),
+		logFile,
+		syslogURI,
+		config.Datadog.GetBool("syslog_rfc"),
+		config.Datadog.GetBool("syslog_tls"),
+		config.Datadog.GetString("syslog_pem"),
+		config.Datadog.GetBool("log_to_console"),
+	)
+	if err != nil {
+		log.Criticalf("Unable to setup logger: %s", err)
+		return nil
+	}
+
+	if confErr != nil {
+		log.Infof("unable to parse Datadog config file, running with env variables: %s", confErr)
+	}
+
+	if !config.Datadog.IsSet("api_key") {
+		log.Critical("no API key configured, exiting")
+		return nil
+	}
+
+	// get hostname
+	hostname, err := util.GetHostname()
+	if err != nil {
+		return log.Errorf("Error while getting hostname, exiting: %v", err)
+	}
+	log.Infof("Hostname is: %s", hostname)
+
+	// start the cmd HTTP server
+	if err = api.StartServer(); err != nil {
+		return log.Errorf("Error while starting api server, exiting: %v", err)
+	}
+
+	// setup the forwarder
+	keysPerDomain, err := config.GetMultipleEndpoints()
+	if err != nil {
+		log.Error("Misconfiguration of agent endpoints: ", err)
+	}
+	f := forwarder.NewDefaultForwarder(keysPerDomain)
+	f.Start()
+	s := &serializer.Serializer{Forwarder: f}
+
+	hname, err := util.GetHostname()
+	if err != nil {
+		log.Warnf("Error getting hostname: %s", err)
+		hname = ""
+	}
+	log.Debugf("Using hostname: %s", hname)
+
+	var metaScheduler *metadata.Scheduler
+	if config.Datadog.GetBool("enable_metadata_collection") {
+		// start metadata collection
+		metaScheduler = metadata.NewScheduler(s, hname)
+
+		// add the host metadata collector
+		// TODO: make it configurable. We shouldn't report them
+		// twice if cluster agent and normal agents are on the same node
+		err = metaScheduler.AddCollector("host", hostMetadataCollectorInterval*time.Second)
```

Comment:
I think the host collector should not be enabled by default. We could think of a edge case where someone wants to run a DCA on a node with no node agent, but we should not support that OOTB.

I'd go with removing it, and for the moment commenting-out the whole metaScheduler part

Reply:
Fair enough, let's take this decision now.